mruby 3.1 以下のときにビルドに失敗する

https://github.com/matsumotory/mruby-localmemcache/pull/7 の変更で入れられた `mruby/internal.h` への参照は mruby が 3.1 以下の場合に解決できないため、 mruby のビルドが失敗するようになりました。

```
#8 69.79 CPP   build/repos/host/mruby-localmemcache/src/lmc_lock.c -> build/host/mrbgems/mruby-cache/src/lmc_lock.pi
#8 69.81 CPP   build/repos/host/mruby-localmemcache/src/lmc_common.c -> build/host/mrbgems/mruby-cache/src/lmc_common.pi
#8 69.83 CPP   build/repos/host/mruby-localmemcache/src/lmc_shm.c -> build/host/mrbgems/mruby-cache/src/lmc_shm.pi
#8 69.84 CPP   build/repos/host/mruby-localmemcache/src/localmemcache.c -> build/host/mrbgems/mruby-cache/src/localmemcache.pi
#8 69.86 CPP   build/repos/host/mruby-localmemcache/src/lmc_error.c -> build/host/mrbgems/mruby-cache/src/lmc_error.pi
#8 69.88 CPP   build/repos/host/mruby-localmemcache/src/mruby_cache_gem.c -> build/host/mrbgems/mruby-cache/src/mruby_cache_gem.pi
#8 69.90 /usr/local/src/ngx_mruby/mruby/build/repos/host/mruby-localmemcache/src/mruby_cache_gem.c:10:10: fatal error: mruby/internal.h: No such file or directory
#8 69.90    10 | #include "mruby/internal.h"
#8 69.90       |          ^~~~~~~~~~~~~~~~~~
#8 69.90 compilation terminated.
#8 69.90 rake aborted!
#8 69.90 Command failed with status (1): [gcc -E -P -std=gnu99 -g -O3 -Wall -Wundef ...]
```

--- a/src/mruby_cache_gem.c
+++ b/src/mruby_cache_gem.c
@@ -7,7 +7,7 @@
 #include "mruby/variable.h"
 #include "mruby/string.h"
 #include "mruby/array.h"
-#if MRUBY_RELEASE_MAJOR >= 3 && MRUBY_RELEASE_MINOR >= 2
+#if MRUBY_RELEASE_MAJOR > 3 || (MRUBY_RELEASE_MAJOR == 3 && MRUBY_RELEASE_MINOR >= 2)
 #include "mruby/internal.h"
 #endif
 #include <sys/types.h>

--- a/src/mruby_cache_gem.c
+++ b/src/mruby_cache_gem.c
@@ -7,7 +7,9 @@
 #include "mruby/variable.h"
 #include "mruby/string.h"
 #include "mruby/array.h"
+#if MRUBY_RELEASE_MAJOR >= 3 && MRUBY_RELEASE_MINOR >= 2
 #include "mruby/internal.h"
+#endif
 #include <sys/types.h>
 #include <unistd.h>
 #include "localmemcache.h"

